core/src/OpenACC/Kokkos_OpenACC_ParallelReduce_Team.hpp

```diff
@@ -372,31 +385,29 @@ KOKKOS_INLINE_FUNCTION void parallel_reduce(
-                                          Functor const& afunctor,         \
-                                          int async_arg) {                 \
-    auto const functor       = afunctor;                                   \
-    auto val                 = aval;                                       \
-    auto const league_size   = policy.league_size();                       \
-    auto const team_size     = policy.team_size();                         \
-    auto const vector_length = policy.impl_vector_length();                \
+#define KOKKOS_IMPL_OPENACC_PARALLEL_REDUCE_DISPATCH_SCHEDULE(REDUCER,  \
+                                                              OPERATOR) \
+  namespace Kokkos::Experimental::Impl {                                \
+  template <class Policy, class ValueType, class Functor>               \
+  void OpenACCParallelReduceTeam##REDUCER(Policy const policy,          \
+                                          ValueType& aval,              \
+                                          Functor const& afunctor,      \
+                                          int async_arg) {              \
+    auto const functor       = afunctor;                                \
+    auto val                 = aval;                                    \
+    auto const league_size   = policy.league_size();                    \
+    auto const team_size     = policy.team_size();                      \
+    auto const vector_length = policy.impl_vector_length();             \
     /* clang-format off */ \
-    KOKKOS_IMPL_ACC_PRAGMA(parallel loop gang num_gangs(league_size) KOKKOS_IMPL_ACC_REDUCE_TEAM_PRAGMA reduction(OPERATOR : val) copyin(functor) async(async_arg))                                               \
-    /* clang-format on */                                                  \
-    for (int i = 0; i < KOKKOS_IMPL_ACC_REDUCE_TEAM_ITRS; i++) {           \
-      int league_id = KOKKOS_IMPL_ACC_REDUCE_TEAM_LEAGUE_ID_INIT;          \
-      typename Policy::member_type team(league_id, league_size, team_size, \
-                                        vector_length);                    \
-      functor(team, val);                                                  \
-    }                                                                      \
-    acc_wait(async_arg);                                                   \
-    aval = val;                                                            \
-  }                                                                        \
+    KOKKOS_IMPL_ACC_PRAGMA(parallel loop gang num_gangs(league_size) KOKKOS_IMPL_ACC_REDUCE_TEAM_PRAGMA reduction(OPERATOR : val) copyin(functor) async(async_arg))                                            \
+    /* clang-format on */                                               \
+    for (int i = 0; i < KOKKOS_IMPL_ACC_REDUCE_TEAM_ITRS; i++) {        \
+      KOKKOS_IMPL_ACC_REDUCE_TEAM_MEMBER_INIT                           \
```

Comment:
Why that change?
I feel like it obfuscate the code

Reply:
It is because different mapping strategies are needed for the host target and the GPU target. 
(`KOKKOS_IMPL_ACC_REDUCE_TEAM_MEMBER_INIT` is defined differently according to the target device type and compiler type.)

Reply:
`KOKKOS_IMPL_ACC_REDUCE_TEAM_MEMBER_INIT` has multiple different definitions depending on the calling contexts.